// ng2-formly/lib/src/ui-bootstrap/types/multicheckbox.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '../../../../src/ui-bootstrap/types/multicheckbox';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/render/api';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/metadata/view';
import * as import5 from '@angular/core/src/linker/view_type';
import * as import6 from '@angular/core/src/change_detection/change_detection';
import * as import7 from '@angular/core/src/linker/component_factory';
import * as import8 from '@angular/core/src/linker/view_container';
import * as import9 from '../../../node_modules/@angular/common/src/directives/ng_for.ngfactory';
import * as import10 from '@angular/core/src/linker/template_ref';
import * as import11 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import12 from '@angular/common/src/directives/ng_for';
import * as import13 from '../../../node_modules/@angular/forms/src/directives/checkbox_value_accessor.ngfactory';
import * as import14 from '../../../node_modules/@angular/forms/src/directives/reactive_directives/form_control_directive.ngfactory';
import * as import15 from '../../../node_modules/@angular/forms/src/directives/ng_control_status.ngfactory';
import * as import16 from '../../../../src/core/services/formly.single.focus.dispatcher';
import * as import17 from '../../core/components/formly.attributes.ngfactory';
import * as import18 from '@angular/core/src/linker/element_ref';
import * as import19 from '@angular/forms/src/directives/checkbox_value_accessor';
import * as import20 from '@angular/forms/src/directives/control_value_accessor';
import * as import21 from '@angular/forms/src/directives/reactive_directives/form_control_directive';
import * as import22 from '@angular/forms/src/directives/ng_control';
import * as import23 from '@angular/forms/src/directives/ng_control_status';
import * as import24 from '../../../../src/core/components/formly.attributes';
export class Wrapper_FormlyFieldMultiCheckbox {
  /*private*/ _eventHandler:Function;
  context:import0.FormlyFieldMultiCheckbox;
  /*private*/ _changed:boolean;
  constructor() {
    this._changed = false;
    this.context = new import0.FormlyFieldMultiCheckbox();
  }
  ngOnDetach(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  ngDoCheck(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import1.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_FormlyFieldMultiCheckbox_Host:import2.RenderComponentType = import3.createRenderComponentType('',0,import4.ViewEncapsulation.None,([] as any[]),{});
class View_FormlyFieldMultiCheckbox_Host0 extends import1.AppView<any> {
  _el_0:any;
  compView_0:import1.AppView<import0.FormlyFieldMultiCheckbox>;
  _FormlyFieldMultiCheckbox_0_3:Wrapper_FormlyFieldMultiCheckbox;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_FormlyFieldMultiCheckbox_Host0,renderType_FormlyFieldMultiCheckbox_Host,import5.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'formly-field-multicheckbox',import3.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_FormlyFieldMultiCheckbox0(this.viewUtils,this,0,this._el_0);
    this._FormlyFieldMultiCheckbox_0_3 = new Wrapper_FormlyFieldMultiCheckbox();
    this.compView_0.create(this._FormlyFieldMultiCheckbox_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import7.ComponentRef_<any>(0,this,this._el_0,this._FormlyFieldMultiCheckbox_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.FormlyFieldMultiCheckbox) && (0 === requestNodeIndex))) { return this._FormlyFieldMultiCheckbox_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._FormlyFieldMultiCheckbox_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.detectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const FormlyFieldMultiCheckboxNgFactory:import7.ComponentFactory<import0.FormlyFieldMultiCheckbox> = new import7.ComponentFactory<import0.FormlyFieldMultiCheckbox>('formly-field-multicheckbox',View_FormlyFieldMultiCheckbox_Host0,import0.FormlyFieldMultiCheckbox);
const styles_FormlyFieldMultiCheckbox:any[] = ([] as any[]);
var renderType_FormlyFieldMultiCheckbox:import2.RenderComponentType = import3.createRenderComponentType('',0,import4.ViewEncapsulation.None,styles_FormlyFieldMultiCheckbox,{});
export class View_FormlyFieldMultiCheckbox0 extends import1.AppView<import0.FormlyFieldMultiCheckbox> {
  _text_0:any;
  _anchor_1:any;
  /*private*/ _vc_1:import8.ViewContainer;
  _TemplateRef_1_5:any;
  _NgFor_1_6:import9.Wrapper_NgFor;
  _text_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_FormlyFieldMultiCheckbox0,renderType_FormlyFieldMultiCheckbox,import5.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this._anchor_1 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._vc_1 = new import8.ViewContainer(1,(null as any),this,this._anchor_1);
    this._TemplateRef_1_5 = new import10.TemplateRef_(this,1,this._anchor_1);
    this._NgFor_1_6 = new import9.Wrapper_NgFor(this._vc_1.vcRef,this._TemplateRef_1_5,this.parentView.injectorGet(import11.IterableDiffers,this.parentIndex),this.ref);
    this._text_2 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._anchor_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import10.TemplateRef) && (1 === requestNodeIndex))) { return this._TemplateRef_1_5; }
    if (((token === import12.NgFor) && (1 === requestNodeIndex))) { return this._NgFor_1_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.to.options;
    this._NgFor_1_6.check_ngForOf(currVal_1_0_0,throwOnChange,false);
    this._NgFor_1_6.ngDoCheck(this,this._anchor_1,throwOnChange);
    this._vc_1.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_1.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import1.AppView<any> {
    if ((nodeIndex == 1)) { return new View_FormlyFieldMultiCheckbox1(this.viewUtils,this,1,this._anchor_1,this._vc_1); }
    return (null as any);
  }
}
class View_FormlyFieldMultiCheckbox1 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _CheckboxControlValueAccessor_4_3:import13.Wrapper_CheckboxControlValueAccessor;
  _NG_VALUE_ACCESSOR_4_4:any[];
  _FormControlDirective_4_5:import14.Wrapper_FormControlDirective;
  _NgControl_4_6:any;
  _NgControlStatus_4_7:import15.Wrapper_NgControlStatus;
  _SingleFocusDispatcher_4_8:import16.SingleFocusDispatcher;
  _FormlyAttributes_4_9:import17.Wrapper_FormlyAttributes;
  _text_5:any;
  _el_6:any;
  _text_7:any;
  _text_8:any;
  /*private*/ _expr_16:any;
  /*private*/ _expr_17:any;
  /*private*/ _expr_18:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import8.ViewContainer) {
    super(View_FormlyFieldMultiCheckbox1,renderType_FormlyFieldMultiCheckbox,import5.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_16 = import6.UNINITIALIZED;
    this._expr_17 = import6.UNINITIALIZED;
    this._expr_18 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','checkbox'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'label',new import3.InlineArray2(2,'class','custom-control custom-checkbox'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n            ',(null as any));
    this._el_4 = import3.createRenderElement(this.renderer,this._el_2,'input',new import3.InlineArray4(4,'class','custom-control-input','type','checkbox'),(null as any));
    this._CheckboxControlValueAccessor_4_3 = new import13.Wrapper_CheckboxControlValueAccessor(this.renderer,new import18.ElementRef(this._el_4));
    this._NG_VALUE_ACCESSOR_4_4 = [this._CheckboxControlValueAccessor_4_3.context];
    this._FormControlDirective_4_5 = new import14.Wrapper_FormControlDirective((null as any),(null as any),this._NG_VALUE_ACCESSOR_4_4);
    this._NgControl_4_6 = this._FormControlDirective_4_5.context;
    this._NgControlStatus_4_7 = new import15.Wrapper_NgControlStatus(this._NgControl_4_6);
    this._SingleFocusDispatcher_4_8 = new import16.SingleFocusDispatcher();
    this._FormlyAttributes_4_9 = new import17.Wrapper_FormlyAttributes(this.renderer,new import18.ElementRef(this._el_4),this._SingleFocusDispatcher_4_8);
    this._text_5 = this.renderer.createText(this._el_2,'',(null as any));
    this._el_6 = import3.createRenderElement(this.renderer,this._el_2,'span',new import3.InlineArray2(2,'class','custom-control-indicator'),(null as any));
    this._text_7 = this.renderer.createText(this._el_2,'\n        ',(null as any));
    this._text_8 = this.renderer.createText(this._el_0,'\n    ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_4,new import3.InlineArray8(6,'change',(null as any),'blur',(null as any),'focus',(null as any)),this.eventHandler(this.handleEvent_4));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._text_8
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import19.CheckboxControlValueAccessor) && (4 === requestNodeIndex))) { return this._CheckboxControlValueAccessor_4_3.context; }
    if (((token === import20.NG_VALUE_ACCESSOR) && (4 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_4_4; }
    if (((token === import21.FormControlDirective) && (4 === requestNodeIndex))) { return this._FormControlDirective_4_5.context; }
    if (((token === import22.NgControl) && (4 === requestNodeIndex))) { return this._NgControl_4_6; }
    if (((token === import23.NgControlStatus) && (4 === requestNodeIndex))) { return this._NgControlStatus_4_7.context; }
    if (((token === import16.SingleFocusDispatcher) && (4 === requestNodeIndex))) { return this._SingleFocusDispatcher_4_8; }
    if (((token === import24.FormlyAttributes) && (4 === requestNodeIndex))) { return this._FormlyAttributes_4_9.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._CheckboxControlValueAccessor_4_3.ngDoCheck(this,this._el_4,throwOnChange);
    const currVal_4_1_0:any = this.parentView.context.formControl.get(this.context.$implicit.key);
    this._FormControlDirective_4_5.check_form(currVal_4_1_0,throwOnChange,false);
    this._FormControlDirective_4_5.ngDoCheck(this,this._el_4,throwOnChange);
    this._NgControlStatus_4_7.ngDoCheck(this,this._el_4,throwOnChange);
    const currVal_4_3_0:any = this.parentView.context.field;
    this._FormlyAttributes_4_9.check_field(currVal_4_3_0,throwOnChange,false);
    const currVal_4_3_1:any = this.parentView.context.formControl.get(this.context.$implicit.key);
    this._FormlyAttributes_4_9.check_formControl(currVal_4_3_1,throwOnChange,false);
    this._FormlyAttributes_4_9.ngDoCheck(this,this._el_4,throwOnChange);
    const currVal_16:any = this.parentView.context.id;
    if (import3.checkBinding(throwOnChange,this._expr_16,currVal_16)) {
      this.renderer.setElementProperty(this._el_4,'id',currVal_16);
      this._expr_16 = currVal_16;
    }
    const currVal_17:any = this.context.$implicit.value;
    if (import3.checkBinding(throwOnChange,this._expr_17,currVal_17)) {
      this.renderer.setElementProperty(this._el_4,'value',currVal_17);
      this._expr_17 = currVal_17;
    }
    this._NgControlStatus_4_7.checkHost(this,this,this._el_4,throwOnChange);
    const currVal_18:any = import3.inlineInterpolate(1,'\n            ',this.context.$implicit.value,'\n            ');
    if (import3.checkBinding(throwOnChange,this._expr_18,currVal_18)) {
      this.renderer.setText(this._text_5,currVal_18);
      this._expr_18 = currVal_18;
    }
  }
  destroyInternal():void {
    this._FormControlDirective_4_5.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_4(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._CheckboxControlValueAccessor_4_3.handleEvent(eventName,$event) && result);
    result = (this._FormlyAttributes_4_9.handleEvent(eventName,$event) && result);
    return result;
  }
}
