// ng2-formly/lib/src/ui-bootstrap/types/textarea.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '../../../../src/ui-bootstrap/types/textarea';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/render/api';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/metadata/view';
import * as import5 from '@angular/core/src/linker/view_type';
import * as import6 from '@angular/core/src/change_detection/change_detection';
import * as import7 from '@angular/core/src/linker/component_factory';
import * as import8 from '../../../node_modules/@angular/forms/src/directives/default_value_accessor.ngfactory';
import * as import9 from '../../../node_modules/@angular/forms/src/directives/reactive_directives/form_control_directive.ngfactory';
import * as import10 from '../../../node_modules/@angular/forms/src/directives/ng_control_status.ngfactory';
import * as import11 from '../../../../src/core/services/formly.single.focus.dispatcher';
import * as import12 from '../../core/components/formly.attributes.ngfactory';
import * as import13 from '@angular/core/src/linker/element_ref';
import * as import14 from '@angular/forms/src/directives/default_value_accessor';
import * as import15 from '@angular/forms/src/directives/control_value_accessor';
import * as import16 from '@angular/forms/src/directives/reactive_directives/form_control_directive';
import * as import17 from '@angular/forms/src/directives/ng_control';
import * as import18 from '@angular/forms/src/directives/ng_control_status';
import * as import19 from '../../../../src/core/components/formly.attributes';
export class Wrapper_FormlyFieldTextArea {
  /*private*/ _eventHandler:Function;
  context:import0.FormlyFieldTextArea;
  /*private*/ _changed:boolean;
  constructor() {
    this._changed = false;
    this.context = new import0.FormlyFieldTextArea();
  }
  ngOnDetach(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  ngDoCheck(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import1.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_FormlyFieldTextArea_Host:import2.RenderComponentType = import3.createRenderComponentType('',0,import4.ViewEncapsulation.None,([] as any[]),{});
class View_FormlyFieldTextArea_Host0 extends import1.AppView<any> {
  _el_0:any;
  compView_0:import1.AppView<import0.FormlyFieldTextArea>;
  _FormlyFieldTextArea_0_3:Wrapper_FormlyFieldTextArea;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_FormlyFieldTextArea_Host0,renderType_FormlyFieldTextArea_Host,import5.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'formly-field-textarea',import3.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_FormlyFieldTextArea0(this.viewUtils,this,0,this._el_0);
    this._FormlyFieldTextArea_0_3 = new Wrapper_FormlyFieldTextArea();
    this.compView_0.create(this._FormlyFieldTextArea_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import7.ComponentRef_<any>(0,this,this._el_0,this._FormlyFieldTextArea_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.FormlyFieldTextArea) && (0 === requestNodeIndex))) { return this._FormlyFieldTextArea_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._FormlyFieldTextArea_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.detectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const FormlyFieldTextAreaNgFactory:import7.ComponentFactory<import0.FormlyFieldTextArea> = new import7.ComponentFactory<import0.FormlyFieldTextArea>('formly-field-textarea',View_FormlyFieldTextArea_Host0,import0.FormlyFieldTextArea);
const styles_FormlyFieldTextArea:any[] = ([] as any[]);
var renderType_FormlyFieldTextArea:import2.RenderComponentType = import3.createRenderComponentType('',0,import4.ViewEncapsulation.None,styles_FormlyFieldTextArea,{});
export class View_FormlyFieldTextArea0 extends import1.AppView<import0.FormlyFieldTextArea> {
  _text_0:any;
  _el_1:any;
  _DefaultValueAccessor_1_3:import8.Wrapper_DefaultValueAccessor;
  _NG_VALUE_ACCESSOR_1_4:any[];
  _FormControlDirective_1_5:import9.Wrapper_FormControlDirective;
  _NgControl_1_6:any;
  _NgControlStatus_1_7:import10.Wrapper_NgControlStatus;
  _SingleFocusDispatcher_1_8:import11.SingleFocusDispatcher;
  _FormlyAttributes_1_9:import12.Wrapper_FormlyAttributes;
  _text_2:any;
  _text_3:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  /*private*/ _expr_14:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_FormlyFieldTextArea0,renderType_FormlyFieldTextArea,import5.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways);
    this._expr_11 = import6.UNINITIALIZED;
    this._expr_12 = import6.UNINITIALIZED;
    this._expr_13 = import6.UNINITIALIZED;
    this._expr_14 = import6.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this._el_1 = import3.createRenderElement(this.renderer,parentRenderNode,'textarea',new import3.InlineArray2(2,'class','form-control'),(null as any));
    this._DefaultValueAccessor_1_3 = new import8.Wrapper_DefaultValueAccessor(this.renderer,new import13.ElementRef(this._el_1));
    this._NG_VALUE_ACCESSOR_1_4 = [this._DefaultValueAccessor_1_3.context];
    this._FormControlDirective_1_5 = new import9.Wrapper_FormControlDirective((null as any),(null as any),this._NG_VALUE_ACCESSOR_1_4);
    this._NgControl_1_6 = this._FormControlDirective_1_5.context;
    this._NgControlStatus_1_7 = new import10.Wrapper_NgControlStatus(this._NgControl_1_6);
    this._SingleFocusDispatcher_1_8 = new import11.SingleFocusDispatcher();
    this._FormlyAttributes_1_9 = new import12.Wrapper_FormlyAttributes(this.renderer,new import13.ElementRef(this._el_1),this._SingleFocusDispatcher_1_8);
    this._text_2 = this.renderer.createText(this._el_1,'    ',(null as any));
    this._text_3 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_1,new import3.InlineArray8(6,'input',(null as any),'blur',(null as any),'focus',(null as any)),this.eventHandler(this.handleEvent_1));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import14.DefaultValueAccessor) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._DefaultValueAccessor_1_3.context; }
    if (((token === import15.NG_VALUE_ACCESSOR) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._NG_VALUE_ACCESSOR_1_4; }
    if (((token === import16.FormControlDirective) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._FormControlDirective_1_5.context; }
    if (((token === import17.NgControl) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._NgControl_1_6; }
    if (((token === import18.NgControlStatus) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._NgControlStatus_1_7.context; }
    if (((token === import11.SingleFocusDispatcher) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._SingleFocusDispatcher_1_8; }
    if (((token === import19.FormlyAttributes) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._FormlyAttributes_1_9.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._DefaultValueAccessor_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_1_0:any = this.context.formControl;
    this._FormControlDirective_1_5.check_form(currVal_1_1_0,throwOnChange,false);
    this._FormControlDirective_1_5.ngDoCheck(this,this._el_1,throwOnChange);
    this._NgControlStatus_1_7.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_1_3_0:any = this.context.field;
    this._FormlyAttributes_1_9.check_field(currVal_1_3_0,throwOnChange,false);
    const currVal_1_3_1:any = this.context.formControl;
    this._FormlyAttributes_1_9.check_formControl(currVal_1_3_1,throwOnChange,false);
    this._FormlyAttributes_1_9.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_11:any = this.context.id;
    if (import3.checkBinding(throwOnChange,this._expr_11,currVal_11)) {
      this.renderer.setElementProperty(this._el_1,'id',currVal_11);
      this._expr_11 = currVal_11;
    }
    const currVal_12:any = this.context.key;
    if (import3.checkBinding(throwOnChange,this._expr_12,currVal_12)) {
      this.renderer.setElementProperty(this._el_1,'name',currVal_12);
      this._expr_12 = currVal_12;
    }
    const currVal_13:any = this.context.to.cols;
    if (import3.checkBinding(throwOnChange,this._expr_13,currVal_13)) {
      this.renderer.setElementProperty(this._el_1,'cols',currVal_13);
      this._expr_13 = currVal_13;
    }
    const currVal_14:any = this.context.to.rows;
    if (import3.checkBinding(throwOnChange,this._expr_14,currVal_14)) {
      this.renderer.setElementProperty(this._el_1,'rows',currVal_14);
      this._expr_14 = currVal_14;
    }
    this._NgControlStatus_1_7.checkHost(this,this,this._el_1,throwOnChange);
  }
  destroyInternal():void {
    this._FormControlDirective_1_5.ngOnDestroy();
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._DefaultValueAccessor_1_3.handleEvent(eventName,$event) && result);
    result = (this._FormlyAttributes_1_9.handleEvent(eventName,$event) && result);
    return result;
  }
}
