// ng2-formly/lib/src/core/services/formly.event.emitter.js
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import { Subject } from 'rxjs/Subject';
export var FormlyValueChangeEvent = (function () {
    function FormlyValueChangeEvent(key, value) {
        this.key = key;
        this.value = value;
    }
    return FormlyValueChangeEvent;
}());
export var FormlyEventEmitter = (function (_super) {
    __extends(FormlyEventEmitter, _super);
    function FormlyEventEmitter() {
        _super.apply(this, arguments);
    }
    FormlyEventEmitter.prototype.emit = function (value) {
        _super.prototype.next.call(this, value);
    };
    return FormlyEventEmitter;
}(Subject));
export var FormlyPubSub = (function () {
    function FormlyPubSub() {
        this.emitters = {};
    }
    FormlyPubSub.prototype.setEmitter = function (key, emitter) {
        this.emitters[key] = emitter;
    };
    FormlyPubSub.prototype.getEmitter = function (key) {
        return this.emitters[key];
    };
    return FormlyPubSub;
}());
